jsjaws.py

```diff
@@ -316,9 +352,89 @@ def execute(self, request: ServiceRequest) -> None:
         # Adding sandbox artifacts using the SandboxOntology helper class
         _ = SandboxOntology.handle_artifacts(self.artifact_list, request)
 
-        # Clean up the modified file, if it exists
-        if path.exists(MODIFIED_SAMPLE):
-            remove(MODIFIED_SAMPLE)
+    def extract_from_soup(self, soup: BeautifulSoup):
+        scripts = soup.findAll("script")
+        aggregated_js_script = None
+        for script in scripts:
+            # Make sure there is actually a body to the script
+            body = script.string
+            if body is None:
+                continue
+            body = str(body).strip()  # Remove whitespace
+            if len(body) <= 2:  # We can treat 2 character scripts as empty
+                continue
+
+            if script.get("type", "").lower() in ["", "text/javascript"]:
+                # If there is no "type" attribute specified in a script element, then the default assumption is
+                # that the body of the element is Javascript
+                encoded_script = body.encode()
+                if aggregated_js_script is None:
+                    aggregated_js_script = tempfile.NamedTemporaryFile(
+                        dir=self.working_directory, delete=False, mode="wb"
+                    )
+                aggregated_js_script.write(encoded_script + b"\n")
+
+        if aggregated_js_script is None:
+            return
+
+        onloads = soup.body.get_attribute_list("onload")
+        for onload in onloads:
+            if onload:
+                aggregated_js_script.write(b"\n" + onload.encode() + b"\n")
+
+        aggregated_js_script.close()
+
+        return aggregated_js_script.name
+
+    def extract_from_html(self, request: ServiceRequest):
+        with open(request.file_path, "rb") as f:
```

Comment:
We are already keeping this value in `file_content`, we shouldn't have to re-read the file.